routes/api/comments/index.js

```diff
@@ -22,34 +26,59 @@ router.get('/:comment_id', (req, res, next) => {
 
 router.post('/', (req, res, next) => {
   let comment  = new Comment({
-    body: req.query.body,
-    author_id: req.query.author_id,
-    asset_id: req.query.asset_id,
-    parent_id: req.query.parent_id,
-    status: req.query.status
+    body: req.body.body,
+    author_id: req.body.author_id,
+    asset_id: req.body.asset_id,
+    parent_id: req.body.parent_id,
+    status: req.body.status
   });
   comment.save().then(({id}) => {
-    res.status(201).send(id);
+    res.status(200).send(id);
   }).catch(error => {
     next(error);
   });
-
 });
 
-router.put('/:comment_id', (req, res) => {
-  res.send('Update a comment');
+router.post('/:comment_id', (req, res, next) => {
+  Comment.findById(req.params.comment_id).then((comment) => {
+    comment.body = req.body.body;
+    comment.author_id = req.body.author_id;
+    comment.asset_id = req.body.asset_id;
+    comment.parent_id = req.body.parent_id;
+    comment.status = req.body.status;
+
+    comment.save().then((comment) => {
+      res.status(200).send(comment);
+    });
+  }).catch(error => {
+    next(error);
+  });
 });
 
-router.delete('/:comment_id', (req, res) => {
-  res.send('Delete a comment');
+router.delete('/:comment_id', (req, res, next) => {
+  Comment.findById(req.params.comment_id).then((comment) => {
+    comment.remove().then(() => {
```

Comment:
add a `return` statement here so that if the remove fails, your `catch` will still work.